cmd/docker/docker.go

```diff
@@ -235,15 +238,20 @@ func hideFeatureSubCommand(subcmd *cobra.Command, hasFeature bool, annotation st
 	}
 }
 
-func hideUnsupportedFeatures(cmd *cobra.Command, details versionDetails) {
+func hideUnsupportedFeatures(cmd *cobra.Command, details versionDetails) error {
 	clientVersion := details.Client().ClientVersion()
 	osType := details.ServerInfo().OSType
 	hasExperimental := details.ServerInfo().HasExperimental
 	hasExperimentalCLI := details.ClientInfo().HasExperimental
+	hasBuildKit, err := details.BuildKitEnabled()
+	if err != nil {
+		return err
+	}
 
 	cmd.Flags().VisitAll(func(f *pflag.Flag) {
 		hideFeatureFlag(f, hasExperimental, "experimental")
 		hideFeatureFlag(f, hasExperimentalCLI, "experimentalCLI")
+		hideFeatureFlag(f, hasBuildKit, "buildkit")
```

Comment:
We can probably remove the `(only if BuildKit enabled)` from these flag descriptions now; https://github.com/docker/cli/blob/ab50c2f2b2de1c912094f97fcf6c1bdd78308412/cli/command/image/build.go#L158-L163


Also in the reference docs; https://github.com/docker/cli/blob/ab50c2f2b2de1c912094f97fcf6c1bdd78308412/docs/reference/commandline/build.md#L51